examples/pincell_random_ray/build_xml.py

```diff
@@ -0,0 +1,203 @@
+      [0.0000000, 0.0000000, 0.0000000, 0.0910284, 0.4155100, 0.0637320, 0.0121390],
+      [0.0000000, 0.0000000, 0.0000000, 0.0000714, 0.1391380, 0.5118200, 0.0612290],
+      [0.0000000, 0.0000000, 0.0000000, 0.0000000, 0.0022157, 0.6999130, 0.5373200],
+      [0.0000000, 0.0000000, 0.0000000, 0.0000000, 0.0000000, 0.1324400, 2.4807000]]])
+scatter_matrix = np.rollaxis(scatter_matrix, 0, 3)
+h2o_xsdata.set_scatter_matrix(scatter_matrix)
+
+mg_cross_sections_file = openmc.MGXSLibrary(groups)
+mg_cross_sections_file.add_xsdatas([uo2_xsdata, h2o_xsdata])
+mg_cross_sections_file.export_to_hdf5()
+
+###############################################################################
+# Create materials for the problem
+
+# Instantiate some Materials and register the appropriate macroscopic data
+uo2 = openmc.Material(name='UO2 fuel')
+uo2.set_density('macro', 1.0)
+uo2.add_macroscopic('UO2')
+
+water = openmc.Material(name='Water')
+water.set_density('macro', 1.0)
+water.add_macroscopic('LWTR')
+
+# Instantiate a Materials collection and export to XML
+materials_file = openmc.Materials([uo2, water])
+materials_file.cross_sections = "mgxs.h5"
+materials_file.export_to_xml()
+
+###############################################################################
+# Define problem geometry
+
+# The geometry we will define a simplified pincell with fuel radius 0.54 cm
+# surrounded by moderator (same as in the multigroup example).
+# In random ray, we typically want several radial regions and azimuthal
+# sectors in both the fuel and moderator areas of the pincell. This is
+# due to the flat source approximation requiring that source regions are
+# small compared to the typical mean free path of a neutron. Below we
+# sudivide the basic pincell into 8 aziumthal sectors (pizza slices) and
+# 5 concentric rings in both the fuel and moderator.
+
+# TODO: When available in OpenMC, use cylindrical lattice instead to
+# simplify definition and improve runtime performance.
+
+pincell_base = openmc.Universe()
+
+# These are the subdivided radii (creating 5 concentric regions in the
+# fuel and moderator)
+ring_radii = [0.241, 0.341, 0.418, 0.482, 0.54, 0.572, 0.612, 0.694, 0.786]
+fills = [uo2, uo2, uo2, uo2, uo2, water, water, water, water, water]
+
+# We then create cells representing the bounded rings, with special
+# treatment for both the innermost and outermost cells
+cells = []
+for r in range(10):
+    cell = []
+    if r == 0:
+        outer_bound = openmc.ZCylinder(r=ring_radii[r])
+        cell = openmc.Cell(fill=fills[r], region=-outer_bound)
+    elif r == 9:
+        inner_bound = openmc.ZCylinder(r=ring_radii[r-1])
+        cell = openmc.Cell(fill=fills[r], region=+inner_bound)
+    else:
+        inner_bound = openmc.ZCylinder(r=ring_radii[r-1])
+        outer_bound = openmc.ZCylinder(r=ring_radii[r])
+        cell = openmc.Cell(fill=fills[r], region=+inner_bound & -outer_bound)
+    pincell_base.add_cell(cell)
+
```

Comment:
That function could partially fulfill what this is doing, but it is really meant for just subdividing the inside of the pin itself. It does not work great for subdividing the moderator region, as you actually don't usually want equal volumes for that part (as you'd like rings to actually be a little more tightly packed close towards the fuel/moderator interface). Additionally, the `openmc.model.pin` function doesn't define any cell in the region of space outside of the last ring, so is kind of awkward. This example could be refactored to use the `model.pin` for the interior and then do some other logic for the outer part, but that seems like it would be more complicated.